plugin/proto/metadata.go

```diff
@@ -42,6 +45,12 @@ func (resp *GetMetadataResponse) Validate() error {
 	if len(resp.SupportedContractVersions) == 0 {
 		return errors.New("empty supported contract versions")
 	}
+	if !resp.SupportsContract(ContractVersion) {
```

Comment:
Should we unexport `Validate()` and call it in the `GetMetadata()` function?

Reply:
Moved Validate() to plugin/plugin.go validate().